dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/TrackedEntityProgramInstanceSupplier.java

```diff
@@ -89,28 +90,34 @@ protected TrackedEntityProgramInstanceSupplier( JdbcTemplate jdbcTemplate )
     @Override
     public void preheatAdd( TrackerImportParams params, TrackerPreheat preheat )
     {
+        if ( params.getEnrollments().size() == 0 )
+            return;
 
         Map<String, List<ProgramInstance>> trackedEntityToProgramInstanceMap = new HashMap<>();
 
         List<String> trackedEntityList = params.getEnrollments().stream().map( Enrollment::getTrackedEntity )
             .collect( Collectors.toList() );
 
+        List<String> programList = params.getEnrollments().stream().map( Enrollment::getProgram ).distinct()
```

Comment:
Maybe here it would be better to get the programs from the Preheat, so we will get only the programs that the user has access to.

Something like this:
preheat.getAll(Program.class)